### PR TITLE
POC/Provisioning: Use healthy repositories

### PR DESCRIPTION
**Pull request by ryantxu**
This PR updates the controller to more accurately keep the health-check up-to-date when *any* configuration changes or enough time has passed.

When requesting repositories, we can now fail quickly if we know they are unhealthy.

### CODE REVIEW

File: pkg/apiserver/registry/generic/strategy.go
Comment (ryantxu) [inline pkg/apiserver/registry/generic/strategy.go:1]: The strategy changes will be incorporated to main from:
https://github.com/grafana/grafana/pull/97946

File: pkg/registry/apis/provisioning/register.go [hunk 1]
```diff
@@ -518,6 +561,13 @@ func (b *ProvisioningAPIBuilder) GetPostStartHooks() (map[string]genericapiserve
 				return err
 			}
 
+			// We do not have a local client until *GetPostStartHooks*, so we can delay init for some
```
Comment (ryantxu) [inline pkg/registry/apis/provisioning/register.go:564]: I moved the tester initialization here so we have access to to the configured client via loopback

File: pkg/apiserver/registry/generic/strategy.go
Comment (ryantxu) [inline pkg/apiserver/registry/generic/strategy.go:1]: See: https://github.com/grafana/grafana/pull/97946

Comment (ryantxu) [reply pkg/apiserver/registry/generic/strategy.go:1]: NOTE -- this is merged into main now

File: pkg/registry/apis/provisioning/repository_controller.go [hunk 2]
```diff
@@ -179,36 +175,61 @@ func (rc *RepositoryController) sync(key string) error {
...
 
+	err = rc.processHealthCheck(context.Background(), cachedRepo)
+	if apierrors.IsNotFound(err) || apierrors.IsConflict(err) {
+		// deleted or changed in the meantime, we'll get called again
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (Proximyst) [inline pkg/registry/apis/provisioning/repository_controller.go:204]:
```suggestion
	return err
```

**Review by Proximyst — APPROVED** [review]
nice!

**Pull request merged**